scipy/sparse/tests/test_base.py

```diff
@@ -4074,6 +4074,17 @@ def test_reshape_copy(self):
         y = x.reshape(new_shape, copy=True)
         assert_(not np.may_share_memory(y.data, x.data))
 
+    def test_large_dimensions(self):
+        # Test that reshape is immune to integer overflow when number of elements
```

Comment:
Not the number of elements, really, but the maximum linear index.

Reply:
I've always envisioned a 4x3 sparse matrix as still being a matrix with 12 elements; it's just that only the nonzero elements take memory. ;-) I'll clarify the comment because that is apparently not universal.